source/common/upstream/cluster_manager_impl.cc

```diff
@@ -530,33 +531,38 @@ ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::ClusterEntry(
                          parent.parent_.local_info_, parent.parent_, parent.parent_.runtime_,
                          parent.parent_.random_,
                          Router::ShadowWriterPtr{new Router::ShadowWriterImpl(parent.parent_)}) {
-
-  switch (cluster->lbType()) {
-  case LoadBalancerType::LeastRequest: {
-    lb_.reset(new LeastRequestLoadBalancer(host_set_, parent.local_host_set_, cluster->stats(),
+  if (cluster->lbSubsetInfo().isEnabled() && cluster->lbType() != LoadBalancerType::OriginalDst) {
```

Comment:
small nit: we should just fail config / doc that subset and original_dst can't be used together.